PYTHON/activityLog/activityLog.py

```diff
@@ -0,0 +1,356 @@
+				serverLoad, thisSlice)
+		logFileName = thisSliceEpoch + logFileName # concatenate strings 
+		return logFileName
+
+	def fileWriteTest(self):
+		try:
+			with open("/testfile.tmp", 'w') as f:
+				f.write("Test file contents")
+				f.close()
+				os.remove("/testfile.tmp")
+				return True
+		except:
+			# print("Cannot write to root (/) directory.")
+			return False
+
+	def getTopOutput(self):
+		# Doesn't work on cygwin, using bottom version in the meantime.
+		# cmd = """top -b -M -H -n1""" 
+		cmd = "top -b -H -n1" 
+		return self.run_command(cmd)[0].decode('utf-8')
+
+	def getNDeviceThroughput(self):
+		cmd = "netstat -i"
+		return self.run_command(cmd)[0].decode('utf-8')
+
+	def getDaemonsAndPorts(self):
+		cmd = "netstat -plunt"
+		return self.run_command(cmd)[0].decode('utf-8')
+
+	def getNetworkConnections(self):
+		cmd = "netstat"
+		return self.run_command(cmd)[0].decode('utf-8')
+
+	def sampleMySQL(self):
+		# TODO: setup error handling for this
+		cmd = "mysql -u{} -p{} --execute \"show full processlist;\"".format(
+			self.mysql_usr, self.mysql_pass)
+		return self.run_command(cmd)[0].decode('utf-8')
+
+	def createTar(self, actLogDir = "~/activityLog"):
+		logcation = "{}/{}".format(actLogDir, self.logfilename)
+		tarlogcation = "{}/{}.tar.gz".format(actLogDir,self.logfilename)
+		try:
+			with tarfile.open(tarlogcation, "w:gz") as tar:
+				tar.add(logcation)
+			os.remove(logcation)
+		except Exception as e:
+			if self.silent_mode == False:
+				print("Error creating tar.gz file.  Exception {}.".format(e))
+
+	def writeTheLog(self, actLogDir = "~/activityLog"):
+		timestamp = self.dateString()
+		actLogDir = self.createLogDir()
+		self.createLogDir(actLogDir)
+		logcation = "{}/{}".format(actLogDir,self.logfilename)
+		try:
+			with open(logcation, 'w') as logfile:
+				logfile.write(self.getTopOutput())
+				logfile.write("\n\n\n\n\n\n\n")
+				logfile.write("Thoroughput on NetWork Interfaces:\n")
+				logfile.write(self.getNDeviceThroughput())
+				logfile.write("\n\n")
+				logfile.write("Daemons and Open Ports list:\n")
+				logfile.write(self.getDaemonsAndPorts())
+				logfile.write("\n\n")
+				logfile.write("Network Connections:\n")
+				logfile.write("\n\n\n\n\n\n\n")
+				# Insert the MySQL bit here
+				if self.mysql_usr != None:
+					for i in range(self.mysql_queries):
+						logfile.write("MySQL Queries Active at {}".format(timestamp))
+						logfile.write("\n")
+						logfile.write(self.sampleMySQL())
+						logfile.write("\n\n")
+						time.sleep(self.mysql_interval)
+		except Exception as e:
+			if self.silent_mode == False:
+				print("Error writing to log file. Exception: {}".format(e))
+
+		# Call createTar() and archive this bitch
+		self.createTar(actLogDir)
+
+
+def main():
+	### Create argument parameters. 
+	# TODO: Find a way to make it case-insensitive.  Something about type = str.lower, 
+	# but I dont know where.
+	parser = argparse.ArgumentParser(
```

Comment:
Good call on using `ArgumentParser`.  This is an essential.   Sometimes it seems like overkill, but I am always glad I went ahead and invested in the overhead somewhere down the line.